Variational_medial_axis/include/CGAL/variational_medial_axis_sampling.h

```diff
@@ -0,0 +1,2181 @@
+    return id;
+  }
+
+  Sphere_ID add_sphere(Sphere_3&& s) {
+    Sphere_ID id = spheres_.emplace(std::move(s));
+    spheres_[id].set_id(id);
+    return id;
+  }
+  MSphere& get_sphere(Sphere_ID idx) {
+    CGAL_precondition(spheres_.owns(idx));
+    return spheres_[idx];
+  }
+
+  const MSphere& get_sphere(Sphere_ID idx) const {
+    CGAL_precondition(spheres_.owns(idx));
+    return spheres_[idx];
+  }
+
+  void remove(Sphere_ID idx) {
+    if(spheres_.owns(idx)) {
+      spheres_.erase(idx);
+    }
+  }
+  void reset() {
+    for(auto& sphere : spheres_) {
+      sphere.reset();
+    }
+  }
+  std::size_t nb_spheres() const { return spheres_.size(); }
+  bool empty() const { return spheres_.empty(); }
+  void clear() { spheres_.clear(); }
+  Sphere_container& spheres() { return spheres_; }
+  const Sphere_container& spheres() const { return spheres_; }
+
+private:
+  Sphere_container spheres_;
+};
+#endif // DOXYGEN_RUNNING
+
+/**  \ingroup PkgVMASRef
+ * \brief Class representing the medial skeleton of a shape.
+ *
+ * This class provides methods to manage and export the medial skeleton of a shape.
+ * It stores the medial spheres, edges, and faces of the skeleton.
+ *
+ * @tparam TriangleMesh The type of the triangle mesh representing the shape.
+ * @tparam GT The geometric traits class used for geometric computations.
+ *         <b>%Default:</b>
+ * \code
+ *     CGAL::Kernel_traits<
+ *       boost::property_traits<
+ *          boost::property_map<TriangleMesh, CGAL::vertex_point_t>::type
+ *        >::value_type
+ *      >::Kernel
+ * \endcode
+ */
+template <typename TriangleMesh_, typename GeomTraits_ = Default> class Medial_Skeleton
+{
+  using GT = typename Default::Get<
+      GeomTraits_,
+      typename Kernel_traits<typename boost::property_traits<
+          typename boost::property_map<TriangleMesh_, vertex_point_t>::type>::value_type>::Kernel>::type;
+  using Sphere_3 = typename GT::Sphere_3;
+  using Point_3 = typename GT::Point_3;
+  using FT = typename GT::FT;
+  using MSMesh = Medial_Sphere_Mesh<TriangleMesh_, GT>;
+  using Sphere_ID = typename MSMesh::Sphere_ID;
+
+public:
+  /**
+   * loads a medial skeleton from a PLY file.
+   *
+   * @param filepath
+   *     Filepath to the PLY file containing the medial skeleton data.
+   * @return
+   *     True if the skeleton was successfully loaded, false otherwise.
+   *
+   * Note: The file format is :
+   * ```
+   * ply
+   * format ascii 1.0
+   * element vertex N
+   * property float x
+   * property float y
+   * property float z
+   * property float radius
+   * element edge M
+   * property int vertex1
+   * property int vertex2
+   * element face K
+   * property list uchar int vertex_indices
+   * end_header
+   * x1 y1 z1 r1
+   * ...          // N vertices
+   * xn yn zn rn
+   * vx vy
+   * ...          // M edges
+   * vz vw
+   * 3 v1 v2 v3
+   * ...          // K faces
+   * 3 vx vy vz
+   * ```
+   */
+  bool load_skeleton_from_ply(std::string& filepath) {
```

Comment:
As we unified IO names into `read_PLY()`  and `write_PLY()`   this should become `read_skeleton_from_PLY()`